### PR TITLE
Improve schema import error propagation

### PR DESCRIPTION
**Pull request by akshay05n**
imodel-native: https://github.com/iTwin/imodel-native/pull/749

### CODE REVIEW

**Review by ColinKerr — APPROVED** [review]

File: core/backend/src/IModelDb.ts
Comment (khanaffan) [inline core/backend/src/IModelDb.ts:888]: Its  `DbResult.BE_SQLITE_ERROR_DataTransformRequired` as it on the master as of now.

File: core/backend/src/IModelDb.ts
Comment (khanaffan) [inline core/backend/src/IModelDb.ts:940]: Same here its `DbResult.BE_SQLITE_ERROR_DataTransformRequired`

**Review by khanaffan — CHANGES_REQUESTED** [review]

File: core/backend/src/IModelDb.ts
Comment (akshay05n) [reply core/backend/src/IModelDb.ts]: Updated. It seems I missed this while resolving merge conflicts.

File: core/backend/src/IModelDb.ts
Comment (akshay05n) [reply core/backend/src/IModelDb.ts]: Updated.

**Review by pmconne — APPROVED** [review]
Pending @khanaffan's approval.

**Comment by pmconne** [comment]
Integration test failure:
```
  1) Schema synchronization
       import schema acquire schema lock when need to transform data:
     Error: Failed to throw error with message: "exclusive lock is already held"
      at assertThrowsAsync (src/integration/SchemaSync.test.ts:108:9)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at Context.<anonymous> (src/integration/SchemaSync.test.ts:930:5)
```

**Comment by akshay05n** [comment]
> Integration test failure:
> 
> ```
>   1) Schema synchronization
>        import schema acquire schema lock when need to transform data:
>      Error: Failed to throw error with message: "exclusive lock is already held"
>       at assertThrowsAsync (src/integration/SchemaSync.test.ts:108:9)
>       at processTicksAndRejections (node:internal/process/task_queues:95:5)
>       at Context.<anonymous> (src/integration/SchemaSync.test.ts:930:5)
> ```

This is due to the change in `importSchemas` and `importXmlSchemas` APIs. The integration test will pass again once the native changes are merged and the new `@bentley/imodeljs-native` package is released.

**Comment by pmconne** [comment]
@tcobbs-bentley @toddsouthenbentley @wgoehrig @aruniverse @calebmshafer your review is required.

**Review by calebmshafer — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Review by toddsouthenbentley — APPROVED** [review]

**Comment by pmconne** [comment]
@DanRod1999 any theories as to why the builds did not trigger when the addon was merged?

**Comment by DanRod1999** [comment]
I will take a closer look. But it might just be a fluke on GitHub. I've seen pipelines not trigger before, but will investigate to make sure it is not a problem in our side

**Pull request merged**
